tasks/tests/unit/test_test_results_processor_task.py

```diff
@@ -0,0 +1,518 @@
+from pathlib import Path
+
+import pytest
+from celery.exceptions import Retry
+from shared.storage.exceptions import FileNotInStorageError
+
+from database.models import CommitReport
+from database.models.reports import Test
+from database.tests.factories import CommitFactory, UploadFactory
+from services.archive import ArchiveService
+from tasks.test_results_processor import TestResultsProcessorTask
+
+here = Path(__file__)
+
+
+class TestUploadTestProcessorTask(object):
+    @pytest.mark.asyncio
+    @pytest.mark.integration
+    async def test_upload_processor_task_call(
+        self,
+        mocker,
+        mock_configuration,
+        dbsession,
+        codecov_vcr,
+        mock_storage,
+        mock_redis,
+        celery_app,
+    ):
+        url = "v4/raw/2019-05-22/C3C4715CA57C910D11D5EB899FC86A7E/4c4e4654ac25037ae869caeb3619d485970b6304/a84d445c-9c1e-434f-8275-f18f1f320f81.txt"
+        with open(here.parent.parent / "samples" / "sample_test.txt") as f:
+            content = f.read()
+            mock_storage.write_file("archive", url, content)
+        upload = UploadFactory.create(storage_path=url)
+        dbsession.add(upload)
+        dbsession.flush()
+        redis_queue = [{"url": url, "upload_pk": upload.id_}]
+        mocker.patch.object(TestResultsProcessorTask, "app", celery_app)
+
+        commit = CommitFactory.create(
+            message="hello world",
+            commitid="cd76b0821854a780b60012aed85af0a8263004ad",
+            repository__owner__unencrypted_oauth_token="test7lk5ndmtqzxlx06rip65nac9c7epqopclnoy",
+            repository__owner__username="joseph-sentry",
+            repository__owner__service="github",
+            repository__name="codecov-demo",
+        )
+        dbsession.add(commit)
+        dbsession.flush()
+        current_report_row = CommitReport(commit_id=commit.id_)
+        dbsession.add(current_report_row)
+        dbsession.flush()
+        result = await TestResultsProcessorTask().run_async(
+            dbsession,
+            repoid=commit.repoid,
+            commitid=commit.commitid,
+            commit_yaml={"codecov": {"max_report_age": False}},
+            arguments_list=redis_queue,
+        )
+        expected_result = {
+            "successful": True,
+            "testrun_list": [
+                {
+                    "duration_seconds": 0.001,
+                    "name": "api.temp.calculator.test_calculator::test_add",
+                    "outcome": "Outcome.Pass",
```

Comment:
Where is this "Outcome.Pass" defined? In the model the field is an Integer, but I can't find the Enum for it....

Reply:
it's defined in the `test-results-parser` library

Reply:
Can we import the type definition into the python code? nice... works like an enum or does it think it's just a string?